data/db.sql

```diff
@@ -1,19 +1,98 @@
-INSERT INTO "user"(firstname, lastname, birthday) VALUES ('Otis', 'Roberson', '1995-01-09');
-INSERT INTO "user"(firstname, lastname, birthday) VALUES ('Jaime', 'King', '1924-05-30');
-INSERT INTO "user"(firstname, lastname, birthday) VALUES ('Vicky', 'Pearson', '1982-12-12)');
-INSERT INTO "user"(firstname, lastname, birthday) VALUES ('Silvia', 'Mcguire', '1971-03-02');
-INSERT INTO "user"(firstname, lastname, birthday) VALUES ('Brendan', 'Pena', '1950-02-17');
-INSERT INTO "user"(firstname, lastname, birthday) VALUES ('Jackie', 'Cohen', '1967-01-27');
-INSERT INTO "user"(firstname, lastname, birthday) VALUES ('Delores', 'Williamson', '1961-07-19');
\ No newline at end of file
+INSERT INTO "user"(firstname, lastname, birthday, pseudo, mail, password, isvalidator) VALUES ('Marwan', 'GUERNOUG', '1997-12-18', 'Kart', 'marwan.guernoug@ensiie.fr', '1234', true);
+INSERT INTO "user"(firstname, lastname, birthday, pseudo, mail, password, isvalidator) VALUES ('Mike', 'MALECOT', '1995-08-06', 'Adolf', 'mike.malecot@ensiie.fr', '1234', true);
+INSERT INTO "user"(firstname, lastname, birthday, pseudo, mail, password, isvalidator) VALUES ('Nicolas', 'CHARLON', '1996-11-09', 'Kozak', 'nicolas.charlon@ensiie.fr', '1234', true);
+INSERT INTO "user"(firstname, lastname, birthday, pseudo, mail, password, isvalidator) VALUES ('Rayan', 'BELMADANI', '1996-10-18', 'Greenns', 'rayan.belmadani@ensiie.fr', '1234', true);
+
+DROP TABLE  "ingredient";
+CREATE TABLE "ingredient" (
+    id SERIAL PRIMARY KEY,
+    label VARCHAR,
+    available BOOLEAN,
+    price REAL,
+    image_link VARCHAR
+);
+
+INSERT INTO "ingredient" (label, available, price)
+VALUES ('Salade', true, 0.2);
+INSERT INTO "ingredient" (label, available, price)
+VALUES ('Tomate', true, 0.3);
+INSERT INTO "ingredient" (label, available, price)
+VALUES ('Oignon', true, 0.3);
+
+DROP TABLE  "sandwich";
+CREATE TABLE "sandwich" (
+    id SERIAL PRIMARY KEY,
+    label VARCHAR
+);
+
+INSERT INTO "sandwich" (label)
+VALUES ('Le kozak gourmand');
+INSERT INTO "sandwich" (label)
+VALUES ('Le marwoille');
+
+DROP TABLE  "sandwich_ingredient";
+CREATE TABLE "sandwich_ingredient" (
+    sandwich_id BIGINT,
+    ingredient_id BIGINT
+);
+
+INSERT INTO "sandwich_ingredient" (sandwich_id, ingredient_id)
+VALUES (1, 1);
+INSERT INTO "sandwich_ingredient" (sandwich_id, ingredient_id)
+VALUES (1, 2);
+INSERT INTO "sandwich_ingredient" (sandwich_id, ingredient_id)
+VALUES (1, 3);
+
+DROP TABLE  "order";
+CREATE TABLE "order" (
+    id SERIAL PRIMARY KEY,
+    order_date DATE,
+    approval BOOLEAN,
+    client_id BIGINT,
+    validator_id BIGINT
+);
+
+INSERT INTO "order" (order_date, approval, client_id, validator_id)
+VALUES ('2020-02-01', true, 1, 2);
+INSERT INTO "order" (order_date, approval, client_id, validator_id)
+VALUES ('2020-02-01', false, 1, 3);
+
+DROP TABLE  "order_sandwich";
+CREATE TABLE "order_sandwich" (
+    order_id BIGINT,
```

Comment:
```suggestion
    order_id BIGINT REFERENCES order(id),
```

Pensez à définir vos foreign key, sinon vous perdez en cohérence de DB.

Aussi, c'est une bonne pratique de définir des index sur les foreign keys, afin d'accélérer le mapping entre vos tables.

Reply:
Pensez aussi à définir une primary key pour chaque table.

Ici, vous pourriez avoir une primary key composite des deux colonnes. Ou bien vous pouvez tout simplement rajouter un id SERIAL primary key

Reply:
Bonjour,
C'est entendu, nous avons tout de même mis des contrôles sur nos services & repo pour protéger un minimum notre db.
Le projet est maj mais les parties corrigées n'ont pas été modifiées.
Cordialement.
SandwicherIIE